build.go

```diff
@@ -139,21 +143,52 @@ func (c *Client) Build(ctx context.Context, opts BuildOptions) error {
 		return err
 	}
 
-	return c.lifecycle.Execute(ctx, build.LifecycleOptions{
+	lifecycleOpts := build.LifecycleOptions{
 		AppPath:            appPath,
 		Image:              imageRef,
 		Builder:            ephemeralBuilder,
 		RunImage:           runImageName,
 		ClearCache:         opts.ClearCache,
 		Publish:            opts.Publish,
+		UseCreator:         false,
+		LifecycleImage:     ephemeralBuilder.Name(),
 		HTTPProxy:          proxyConfig.HTTPProxy,
 		HTTPSProxy:         proxyConfig.HTTPSProxy,
 		NoProxy:            proxyConfig.NoProxy,
 		Network:            opts.ContainerConfig.Network,
 		Volumes:            platformVolumes,
 		DefaultProcessType: opts.DefaultProcessType,
 		FileFilter:         opts.FileFilter,
-	})
+	}
+
+	// Technically the creator is supported as of platform API version 0.3 (lifecycle version 0.7.0+) but earlier versions
+	// have bugs that make using the creator problematic.
+	lifecycleSupportsCreator := !ephemeralBuilder.LifecycleDescriptor().Info.Version.LessThan(semver.MustParse("0.7.4"))
+
+	if lifecycleSupportsCreator && (!opts.Publish || opts.TrustBuilder) {
+		// no need to fetch a lifecycle image, it won't be used
+		lifecycleOpts.UseCreator = true
+		return c.lifecycle.Execute(ctx, lifecycleOpts)
+	}
+
+	lifecycleImageSupported := !ephemeralBuilder.LifecycleDescriptor().Info.Version.LessThan(semver.MustParse("0.7.5"))
+	if !lifecycleImageSupported {
+		c.logger.Warnf("Lifecycle does not have an associated lifecycle image (%s). Each lifecycle phase will be run in a separate container using the provided builder image. Registry credentials will be provided only to analyze, restore, and export.\nRun `pack build` with `--trust-builder` to silence this warning.", lifecycleImageRepo)
```

Comment:
1. The details about it running in separate containers and credentials seem to be too verbose.
2. We shouldn't warn about `pack` CLI specific flags within this context. This code path can be executed by using pack as a Go library.

Probably not the best suggestion but something like this?

```suggestion
		c.logger.Warn("Lifecycle version %s does not have an associated lifecycle image (%s).", ephemeralBuilder.LifecycleDescriptor().Info.Version, lifecycleImageRepo)
```

The suggestion to use `--trust-builder` should be elsewhere in the `internal/commands` or `cmd` packages.

Reply:
Thank you! I made the change.

Reply:
Thanks for the update, I missed the part about adding the provided lifecycle version. The current message is much clearer.